src/app/public/modules/ag-grid/ag-grid-data-manager-adapter.directive.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  AfterContentInit,
+  ChangeDetectorRef,
+  ContentChildren,
+  Directive,
+  Input,
+  OnDestroy,
+  QueryList
+} from '@angular/core';
+
+import {
+  Subject,
+  Subscription
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  AgGridAngular
+} from 'ag-grid-angular';
+
+import {
+  ColumnMovedEvent,
+  RowSelectedEvent
+} from 'ag-grid-community';
+
+import {
+  SkyDataManagerService,
+  SkyDataManagerSortOption,
+  SkyDataManagerState,
+  SkyDataViewConfig
+} from '@skyux/data-manager';
+
+@Directive({
+  selector: '[skyAgGridDataManagerAdapter]'
+})
+export class SkyAgGridDataManagerAdapterDirective implements AfterContentInit, OnDestroy {
+
+  @Input()
+  private viewId: string;
+
+  @ContentChildren(AgGridAngular, { descendants: true })
+  public agGridList: QueryList<AgGridAngular>;
+
+  private currentAgGrid: AgGridAngular;
+  private dataStateSub: Subscription;
+  private viewConfig: SkyDataViewConfig;
+  private _ngUnsubscribe = new Subject();
```

Comment:
We typically only use underscores on private properties when they're used to store the state of public `get`ter/`set`er properties.